src/notion-provider.ts

```diff
@@ -33,82 +50,183 @@ function NotionProvider(this: any, options: NotionProviderOptions) {
+
+              return entize(res)
             }
           },
-                    
-	  save: {
-	    action: async function(this: any, entize: any, msg: any) {
-	      let ent = msg.ent
-	      let id = ent.id
-	      let properties = ent.properties
-	      let obj
-	      try{
-	        await this.shared.sdk.pages.update({page_id: id, properties: ent.properties})
-		// obj = await this.entity('provider/notion/page').load$(id) // a fix to get all the properties
-	      }
-	      catch(err){
-	        if(err.status >= 400 && err.status < 500){
-		  return null
-		}
-		throw err
-	      }
-	      return ent // a more efficient fix for the properties issue - less efficient: uncomment "obj" and return entize(obj)
+
+          save: {
+            action: async function(this: any, entize: any, msg: any) {
+              let q: any = msg.q || {}
+              let ent: any = msg.ent
+              let id: any = ent.id
+              let properties: any = ent.properties || {}
+              let title: any = ent.title || []
+              let res: any
+
+              (!q.page_id && !id) ? this.fail('invalid_page_id') : null
+
+              const config = null == id
+                ? {
+                    method: 'POST',
+                    body: {
+                      'parent': {
+                        'page_id': q.page_id
+                      },
+                      'title': [ ...title ],
+                      'properties': { ...properties }
+                    }
+                  }
+                : {
+                    method: 'PATCH',
+                    body: JSON.stringify({
+                      'title': [ ...title ],
+                      'properties': { ...properties }
+                    })
+
+                  };
+              (null == id)
+                ? ( res = await postJSON('https://api.notion.com/v1/databases',
+                makeConfig(config)) )
+                : (res = await fetch(`https://api.notion.com/v1/databases/${id}`, makeConfig(config)),
+                   res = await res.json() )
+
+
+              return entize(res)
+
             }
+
           }
+
+
         }
+
       }
+      
     }
-  });
+
+  })
     
   this.prepare(async function(this: any) {
-    let authTok = await this.post('sys:provider,get:key,provider:notion,key:authToken')
-    if (!authTok.ok) {
-      this.fail('api-key-missing')
+    let seneca = this
+
+    let res =
+      await seneca.post('sys:provider,get:keymap,provider:notion')
+
+    if (!res.ok) {
+      this.fail('notion-missing-keymap', res)
     }
-    this.shared.sdk = new Notion.Client({ auth: authTok.value })
-    this.shared.authTok = authTok.value // needed for 'fetch'
+
+    let authToken = res.keymap.authToken.value
+
+    // TODO: authenticate
+    seneca.shared.headers = {
+      'Authorization': `Bearer ${authToken}`,
+      'Accept': 'application/json',
+      'Notion-Version': '2021-05-13',
+      'Content-Type': 'application/json'
+    }
```

Comment:
add a `...options.headers` here to allow user to override

Reply:
yes - done!